src/main/java/com/fasterxml/jackson/dataformat/xml/ser/ToXmlGenerator.java

```diff
@@ -475,6 +508,25 @@ public final void writeFieldName(String name) throws IOException
             _reportError("Can not write a field name, expecting a value");
         }
         // Should this ever get called?
+        _nextRealName = null;
+        if(!VALID_XML_TAG.matcher(name).matches()) {
+            switch (_tagEscapingStrategy) {
+                case ATTRIBUTE_ESCAPE:
+                    _nextRealName = name;
+                    name = ESCAPED_TAG_NAME;
+                    break;
+                case REPLACE:
+                    name = name.replaceAll("^[^a-zA-Z_:]", "_");
```

Comment:
Very inefficient, constructing Regexp every time is called. Also unnecessarily replaces numbers, f.ex.
(but then again, this is not my main concern: the whole processing even with pre-constructed RE is rather expensive compared to doing this at databind).

Reply:
Sure, I will do that (assuming this is still relevant). However, regarding numbers: This should only match characters at the beginning of the string, where numbers are not allowed (that's why there are 2 `replaceAll`).